### DISCUSSION TITLE
Main PCB source

### DISCUSSION THREAD

**Discussion by mredmon63** [nasa-jpl/open-source-rover#258] [Q&A]
Does any company stock the main PCB or is the only way to source one is through a board shop by sending the gerber files?  I would like to purchase one fully populated board if that is possible.
Thank you,
Mark
Melbourne FL

**Comment by ericjunkins** [comment]
Hi,

Sorry there is no way to buy a fully assembled board, unless you can convince one of them community members to build one for you :p

**Comment by mredmon63** [comment]
Thanks for the reply.  No problem, I can hand solder a board.

What is the best way to purchase a bare pcb?

**Comment by SConaway** [comment]
@mredmon63 you can purchase one from a fab like OSHPark (high quality but super expensive) or JLCPCB (much cheaper but worse quality). I've ordered several boards through JLCPCB and they've been fine, but YMMV.

**Comment by mredmon63** [comment]
Thank you Steven.
To order a board from a fab shop, do I just need to provide all the files in the ...pcb/control_board/gerbers/rev_e folder?

I appreciate you helping me through the control board acquisition.

**Comment by SConaway** [comment]
Glad to help, Mark! 

That would work, but since most places expect a `.zip` to be uploaded, you can download the `Control Board.zip` from `electrical/pcb/Control Board.zip` and then just upload / provide them with that.

**Comment by mredmon63** [comment]
You were correct, OshPark was expecting a zip.
The zip appeared to contain rev B and rev F files.  Are you confident OshPark will manufacture the latest version of the Control Board?
If so I will purchase their minimum three boards for $210 and put two up for sale on the GitHub forum.
Thanks,
Mark

**Comment by SConaway** [comment]
As far as I know, both sets of files are needed for the PCB. I just uploaded them to OshPark and JLCPCB and they seemed to figure it out just fine. 

Not sure what your timeline is like, but if you're willing to wait a few extra days (up to a week), JLCPCB is about $50 for 5 boards (their minimum).

**Comment by apollokit** [comment]
@mredmon63  Make sure you've looked at the instructions in https://github.com/nasa-jpl/open-source-rover/tree/master/electrical, that should clarify some of this. Also make sure you're using the settings mentioned there.

Those aren't rev B and F in Control Board.zip - that's the Front and Back layer. So those files are correct, for rev E of the control board

I would definitely consider ordering from JLCBCP like @SConaway said, $210 is a lot. And note that we're currently working on the next rev of the board (F), so within a couple months you might want to order a board again (if you want to update, that is).

**Comment by talueder** [comment]
I am trying to order on JLCPCB, and it now asks for a CPL file.  Where would I find this in the documentation?  TIA

<img width="758" alt="image" src="https://user-images.githubusercontent.com/125751060/220140666-98083111-8b71-4bcf-b3a5-3c4f2c74b4f7.png">
